sdk/provisioning/Azure.Provisioning.Dns/src/DnsTxtRecord.cs

```diff
@@ -91,11 +91,11 @@ public DnsTxtRecord(string bicepIdentifier, string? resourceVersion = default)
     protected override void DefineProvisionableProperties()
     {
         base.DefineProvisionableProperties();
-        _name = DefineProperty<string>("Name", ["name"], isRequired: true);
-        _id = DefineProperty<ResourceIdentifier>("Id", ["id"], isOutput: true);
-        _systemData = DefineModelProperty<SystemData>("SystemData", ["systemData"], isOutput: true);
-        _txtRecords = DefineListProperty<DnsTxtRecordInfo>("TxtRecords", ["properties", "TXTRecords"]);
-        _parent = DefineResource<DnsZone>("Parent", ["parent"], isRequired: true);
+        _name = DefineProperty<string>(nameof(Name), ["name"], isRequired: true);
+        _id = DefineProperty<ResourceIdentifier>(nameof(Id), ["id"], isOutput: true);
+        _systemData = DefineModelProperty<SystemData>(nameof(SystemData), ["systemData"], isOutput: true);
+        _txtRecords = DefineListProperty<DnsTxtRecordInfo>(nameof(TxtRecords), ["properties", "TXTRecords"]);
```

Comment:
I wonder if we could simplify this definition and skip the definition of `TxtRecords` and have, similar to what we did to `CName`, this inlined as `_txtRecords = DefineListProperty<DnsTxtRecordInfo>(nameof(TxtRecords), ["properties", "TXTRecords", "Values"]);`

Reply:
usually we do not want to do too many modifications to the properties other than flatten stuff comparing to their real json payload.
changing them has a risk that in the future the service team might update the API and put us in an awkward situation.